From 03671727694984b53a1c17024ed738a700d0a7f4 Mon Sep 17 00:00:00 2001
From: Romain Pokrzywka <romain.pokrzywka@kdab.com>
Date: Mon, 16 Jan 2012 01:17:06 -0600
Subject: Fix duplicate mouse event being sent when single touch events are triggered.
 This is due to the X and Y messages being received before the touch,
 followed by a SYNC which then generates a second mouseEvent with posChanged=true.

---
 src/plugins/generic/linuxinput/qlinuxinput.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/plugins/generic/linuxinput/qlinuxinput.cpp b/src/plugins/generic/linuxinput/qlinuxinput.cpp
index aa2eccc..8bbdbf3 100644
--- a/src/plugins/generic/linuxinput/qlinuxinput.cpp
+++ b/src/plugins/generic/linuxinput/qlinuxinput.cpp
@@ -277,7 +277,7 @@ void QLinuxInputMouseHandler::readMouseData()
         } else if (data->type == EV_KEY && data->code == BTN_TOUCH) {
             m_buttons = data->value ? Qt::LeftButton : Qt::NoButton;
 
-            sendMouseEvent(m_x, m_y, m_buttons);
+            posChanged = true;
             pendingMouseEvent = false;
         } else if (data->type == EV_KEY && data->code >= BTN_LEFT && data->code <= BTN_MIDDLE) {
             Qt::MouseButton button = Qt::NoButton;
-- 
1.7.4.1

